FEM/rf_st_new.cpp

```diff
@@ -3892,12 +3883,39 @@ void CSourceTermGroup::SetSurfaceNodeValueVector(CSourceTerm* st, Surface* m_sfc
 	}
 }
 
+#if defined(USE_PETSC) // || defined (other parallel linear solver lib). //WW. 06.2016
+void CSourceTermGroup::removeGhostElements(std::vector<long>& node_ids,
+                                        std::vector<double>& node_valus)
+{	const std::size_t id_act_l_max = static_cast<std::size_t>(m_msh->getNumNodesLocal());
+	const std::size_t id_max_l = m_msh->GetNodesNumber(false);
+	const std::size_t id_act_h_max = m_msh->getLargestActiveNodeID_Quadratic();
+
+	std::vector<std::size_t> ghost_ids;
+	for (std::size_t i=0; i<node_ids.size(); i++)
+	{
+		const std::size_t id = static_cast<std::size_t>(node_ids[i]);
+		// non ghost element
+		if (   (id < id_act_l_max)
```

Comment:
For readability and reusing, i think it's better to move this checking to Mesh class, i.e. making a new member function something like `isNodeInGhostElement()`?